apps/main/src/dex/components/PageCreatePool/hooks/useOracleValidation.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react'
+import { isAddress, type Address } from 'viem'
+import { useOracleRate } from '@/dex/components/PageCreatePool/hooks/useOracleRate'
+import { TokenId, TokenState } from '@/dex/components/PageCreatePool/types'
+import { validateOracleFunction } from '@/dex/components/PageCreatePool/utils'
+import useStore from '@/dex/store/useStore'
+
+type UseOracleValidationParams = {
+  token: TokenState
+  tokenId: TokenId
+}
+
+/**
+ * This hook validates oracle configuration, syncs results to store, and returns them
+ */
+export const useOracleValidation = ({ token, tokenId }: UseOracleValidationParams) => {
+  const updateOracleState = useStore((state) => state.createPool.updateOracleState)
+
+  const oracleAddress = token.oracle.address
+  const oracleFunction = token.oracle.functionName
+
+  const enabled = isAddress(oracleAddress) && oracleFunction !== '' && validateOracleFunction(oracleFunction)
```

Comment:
The `!== ''` isn't needed right? It's implicit from the `validateOracleFunction` check